module/zfs/arc.c

```diff
@@ -4071,25 +4117,108 @@ arc_evict_state(arc_state_t *state, arc_buf_contents_t type, uint64_t spa,
 		multilist_sublist_unlock(mls);
 	}
 
+	evict_arg_t *evarg = kmem_alloc(sizeof (*evarg) * num_sublists,
+	    KM_SLEEP);
```

Comment:
Sleepable memory allocation in eviction path is a request for potential troubles.

Reply:
I've changed that to NOSLEEP. Now, if we cannot allocate the memory, we just fall back to the regular single evict.

Reply:
Thanks. That is one way to solve it. Or we could pre-allocate it similar to markers.